src/main/java/com/fasterxml/jackson/databind/introspect/JacksonAnnotationIntrospector.java

```diff
@@ -1547,6 +1547,8 @@ protected TypeResolverBuilder<?> _findTypeResolver(MapperConfig<?> config,
             b = b.defaultImpl(defaultImpl);
         }
         b = b.typeIdVisibility(info.visible());
+        // [databind#3877]: allow configuration of per-type strict type handling
+        b.requireTypeIdForSubtypes(info.requireTypeIdForSubtypes().asBoolean());
```

Comment:
The point of `OptBoolean` is to have "third option" of "not specified", so I think this should check that case (skip call if value is `OptBoolean.DEFAULT`)

Reply:
Oh. Actually, never mind. `asBoolean()` returns `null` for the case, so I think this is fine.

Reply:
add a comment though